### PR TITLE
MSA-786: Encored Technologies : Smart Energy Service

### PR DESCRIPTION
**Pull request by hhhss91**
Submitting 'Encored Technologies : Smart Energy Service' for publication

### CODE REVIEW

File: devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "EnerTalk Energy Meter", namespace: "Encored Technologies", author: "hyeon seok yang") {
+	}
+
+	simulator {
+		// TODO: define status and reply messages here
```
Comment (jasonbio) [inline devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy:21]: Empty simulator - please define

File: devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+       		state "tier", label:' ${currentValue}'
+		}
+        valueTile("plan", "device.plan", width: 2, height : 2, decoration: "flat") {
+       		state "plan", label:' ${currentValue}'
+		}
+        
+        htmlTile(name:"deepLink", action:"linkApp", whitelist:["code.jquery.com", 
+    	"ajax.googleapis.com", 
+        "fonts.googleapis.com",
+    	"code.highcharts.com",
```
Comment (jasonbio) [inline devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy:44]: "code.highcharts.com" - can this JS be stored on a more reliable domain such as AWS? Is this even necessary?

File: smartapps/encored-technologies/smart-energy-service.src/smart-energy-service.groovy [hunk 3]
```diff
@@ -0,0 +1,1388 @@
...
+    card(name: "Encored Energy Service", type: "html", action: "getHtml", whitelist: whiteList()) {}
+}
+
+/* This list contains, that url need to be allowed in Smart Energy Service.*/
+def whiteList() {
+	[
+    	"code.jquery.com", 
+    	"ajax.googleapis.com", 
+        "fonts.googleapis.com",
+    	"code.highcharts.com",
```
Comment (jasonbio) [inline smartapps/encored-technologies/smart-energy-service.src/smart-energy-service.groovy:48]: "code.highcharts.com" - can this JS be stored on a more reliable domain such as AWS? Is this even necessary?

File: devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy (hunk 1 above)
Comment (hhhss91) [inline devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy:21]: I am not sure what should go in there?

File: devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy (hunk 2 above)
Comment (hhhss91) [inline devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy:44]: I will remove it, I think I just forgot to delete that. It's no longer used.

File: devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy (hunk 1 above)
Comment (jasonbio) [inline devicetypes/encored-technologies/enertalk-energy-meter.src/enertalk-energy-meter.groovy:21]: You can just remove it if you don't intend to define any simulator settings - they're not required

**Comment by jasonbio** [comment]
One final request - if possible can you please clean up the tabbing/spacing on the code? This will make it easier to navigate in the future.

File: smartapps/encored-technologies/smart-energy-service.src/smart-energy-service.groovy [hunk 4]
```diff
@@ -0,0 +1,1388 @@
...
+    	
+        log.debug "Encored Access Token does not exist."
+        
+        if (!state.accessToken) { /*if smartThings' access token does not exitst*/
+        	log.debug "SmartThings Access Token does not exist."
+            
+            createAccessToken() /*request and get access token from smartThings*/
+            
+            /* re-create strings to make sure it's been initialized. */
+            //createLocaleStrings()
```
Comment (jasonbio) [inline smartapps/encored-technologies/smart-energy-service.src/smart-energy-service.groovy:106]: Please remove unused code

**Comment by jasonbio** [comment]
This PR is being merged specifically for deployment in AP01 due to partner request (DVCSMP-1425)

**Pull request merged**